src/NServiceBus.AcceptanceTests/DelayedDelivery/When_using_custom_delayed_delivery.cs

```diff
@@ -0,0 +1,146 @@
+﻿namespace NServiceBus.AcceptanceTests.DelayedDelivery
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using DeliveryConstraints;
+    using EndpointTemplates;
+    using Extensibility;
+    using Features;
+    using NServiceBus.DelayedDelivery;
+    using NServiceBus.Performance.TimeToBeReceived;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+    using Transport;
+
+    public class When_using_custom_delayed_delivery : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_replace_the_default_behavior()
+        {
+            var delay = TimeSpan.FromMilliseconds(1);
+
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b => b.When((session, c) =>
+                {
+                    var options = new SendOptions();
+
+                    options.DelayDeliveryWith(delay);
+                    options.RouteToThisEndpoint();
+
+                    return session.Send(new MyMessage(), options);
+                }))
+                .Done(c => c.WasCalled)
+                .Run();
+
+            Assert.IsTrue(context.WasCalled);
+        }
+
+        public class Context : ScenarioContext
+        {
+            public bool WasCalled { get; set; }
+        }
+
+        public class Endpoint : EndpointConfigurationBuilder
+        {
+            public Endpoint()
+            {
+                EndpointSetup<DefaultServer>(config =>
+                {
+                    config.DisableFeature<TimeoutManager>();
+                    config.EnableFeature<CustomTimeoutManager>();
+                });
+            }
+
+            class CustomTimeoutManager : Feature
+            {
+                protected override void Setup(FeatureConfigurationContext context)
+                {
+                    var satelliteLogicalAddress = context.Settings.LogicalAddress().CreateQualifiedAddress("CustomTM");
+                    var satelliteAddress = context.Settings.GetTransportAddress(satelliteLogicalAddress);
+
+                    context.AddSatelliteReceiver("CustomTM", satelliteAddress, TransportTransactionMode.ReceiveOnly, PushRuntimeSettings.Default, RecoverabilityPolicy, (builder, messageContext) =>
+                    {
+                        var ctx = builder.Build<Context>();
+                        ctx.WasCalled = true;
+                        return Task.FromResult(0);
+                    });
+
+                    context.Pipeline.Replace("ThrowIfCannotDeferMessage", new RouteDeferredMessageToTimeoutManagerBehavior(satelliteAddress));
```

Comment:
I agree with @timbussmann let's remove the `When_using_custom_delayed_delivery ` test.